server/src/controllers/v1/auth.js

```diff
@@ -0,0 +1,93 @@
+import User from '../../models/user';
+import { newToken } from '../../utils/helpers/jwtHelper';
+import errorMessage from '../../utils/errorMessage';
+import statusMessage from '../../utils/statusMessage';
+import destructureUser from '../../utils/destructureUser';
+
+const controllerStatus = (req, res) => res.status(200).json({ status: 'online' });
+
+const signinController = async (req, res) => {
+  let user;
+
+  try {
+    user = await User.findOne({
+      email: req.body.email,
+    }).select('+password');
+
+    if (!user) {
+      throw Error;
+    }
+  } catch (error) {
+    return errorMessage({
+      status: 400,
+      message: 'Email not found. Please sign up',
+    }, req, res);
+  }
+
+  try {
+    const isMatch = await user.checkPassword(req.body.password);
+
+    user = user.toObject();
+    delete user.password;
+
+    if (isMatch) {
+      const userData = destructureUser(user);
+      const token = newToken({ ...userData });
+      return statusMessage({
+        success: true,
+        status: 200,
+        message: 'Sign-in successful',
+        user: { ...userData, token },
+      }, req, res);
+    }
+    throw Error;
+  } catch (error) {
+    return errorMessage({
+      status: 400,
+      message: 'Sign-in failed. Check email and/or password',
+    }, req, res);
+  }
+};
+
+const signupController = async (req, res) => {
+  try {
+    const user = await User.create(req.body);
```

Comment:
line breaks please, it's very unreadbable

Reply:
ESLint automatically removes the first empty line after start of a block